app/src/stores/post.ts

```diff
@@ -42,7 +43,22 @@ export const usePostStore = defineStore("post", () => {
             return new Promise((resolve) => {
                 resolve(
                     posts.filter((post) => {
-                        return post.content[0]?.status == ContentStatus.Published;
+                        const isPublished = post.content[0]?.status == ContentStatus.Published;
```

Comment:
I think there should still be an early return here, `if (!isPublished) return false` => because we only want to display published posts anyways so then we already know that this one should not be visible.

Reply:
Here is the return: `return isPublished && isNotExpired && isAvailable;`

Reply:
Yes but I mean check for `isPublished` already at the top of the method, that is called an "early return", because you're not running any of the other code (potentially expensive operation). We can already exclude a lot of content by first checking if it's published at all (currently non-published posts get sent to the client as well, we might change that in the future).